From 35d0248e06ba2ded776e9649bf152b7088f0c698 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Vesa=20J=C3=A4=C3=A4skel=C3=A4inen?=
 <vesa.jaaskelainen@vaisala.com>
Date: Thu, 8 Apr 2021 16:23:26 +0300
Subject: [PATCH 7/9] libckteec: fix deserialization for CKA_ALLOWED_MECHANISMS
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add special deserialization support for CKA_ALLOWED_MECHANISMS.

Signed-off-by: Vesa Jääskeläinen <vesa.jaaskelainen@vaisala.com>
---
 libckteec/src/serialize_ck.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libckteec/src/serialize_ck.c b/libckteec/src/serialize_ck.c
index 716c7ee..f114550 100644
--- a/libckteec/src/serialize_ck.c
+++ b/libckteec/src/serialize_ck.c
@@ -344,6 +344,9 @@ CK_RV deserialize_ck_attributes(uint8_t *in, CK_ATTRIBUTE_PTR attributes,
 		if (cur_attr->pValue) {
 			if (ck_attr_is_ulong(cur_attr->type))
 				len += sizeof(uint32_t);
+			else if (cur_attr->type == CKA_ALLOWED_MECHANISMS)
+				len += cur_attr->ulValueLen / sizeof(CK_ULONG) *
+				       sizeof(uint32_t);
 			else
 				len += cur_attr->ulValueLen;
 		}
-- 
2.25.1

